### DISCUSSION TITLE
Markdown Preview & Thoughts as a New User

### DISCUSSION THREAD

**Discussion by thwaller** [Zettlr/Zettlr#2719] [Ideas]
I read a few prior posts on the depreciated forum on this topic, and I understand that the response is that Zettlr is a WYSIWTG software. However... I would like to comment on this....

I am a new user of Zettlr, only having started using it for approx 1 week now. I have used, and still use, numerous other softwares that are similar, MD editors. I disagree with the above statement. When I type in Zettlr, I see for example a "#" in front of my headers. That s not "what I get" in the final result, making it **not** a WYSIWYG editor. What I see is the "code" and not the actual result, which is what a WYSIWYG editor does.

I have been using markdown for a good amount of time now, and I have experienced issues with the many different flavors of it. This platform does this and this software does that... it is not all the same and not consistent... although usually close. It would be great to see a preview of what you have coded so you can see how it will generate without needing to do a temp export to see it. I understand one of the responses that a split screen is a waste of real estate, but simply offering a button to offer a preview would go a long way to resolving the issue.

My main editor is Joplin, since it uses a DB to store data. However, there are many times that I simply need an editor to create and edit plain old MD files. Typora and Ghostwriter have typically been my go-to softwares, Typora being a sore spot as it is not open source. Both a great, as is Zettlr, but I believe that all have their pros and cons. What I am looking for is sort of a LibreOffice Writer but using MD vs the standard DOCX/OTD. I believe MD is the new document format, with exception to maybe reports where you might want ODT or statistical docs where you might need RMD or LaTeX.

I believe that Zettlr is the closest software I have found to this goal. I believe that if the development takes more of an open mind this could be something big, and I would be happy to help it to get there and use it during the process. It sort of bothers me to read the responses to the preview requests though... it is never a bad thing to be able to see what the result of your code will look like. Using other softwares, IDEs, like VSCode, R-Studio and Atom... they all offer an option to preview what you have done. I really believe that such a feature along with other enhancements would make Zettlr the best MD editor to have.

**Comment by RoyiAvital** [comment]
I would also like to have side by side mode with RAW text on the left / right and rendered output on the right / left.

This WYSIWYG isn't efficient when writing technical documents with a lot of math.

**Comment by freepicheep** [comment]
I agree, @thwaller. In the mean time, you should check out [MarkText](https://github.com/marktext/marktext). It is a FOSS that even supports live preview (similar to Typora) of markdown files. It's not as powerful as Zettlr, but it might satisfy some of your requirements for a markdown editor.

**Reply by Pomax** [reply]
MarkText unfortunately cannot deal with "large documents", to the point where its undo/redo is completely useless because it takes way too long to kick in, and it will happily skip to "the start of the document" when it you tell it to skip to the start of a line. I would _love_ to switch away from Typora for MarkText, but it's just not good as a true document editor, only as a "small document" editor.

**Comment by thwaller** [comment]
@aCodingLegolas I have Mark Text installed and do use it, however, as you pointed out, it is somewhat lacking. I used to use Typora a LOT, but since their change in licensing shifting from proprietary but free to proprietary paid only, I left. They used all of us in open source to help them develop, now they want to charge without even fixing all of the reported issues... sad.

I am using Obsidian for most MD now. It is a bit bulky compared to a simple MD editor like Typora, Mark Text, etc and is somewhat similar to Zettlr in that it also acts as a MD organizer/browser of sorts in addition to an editor. It has a lot of great features and you can easily switch between reading, source and live views.

I also use Ghostwriter, Gedit (with MD plugin) and others depending on the needs at the time.

**Comment by nasteffe** [comment]
To be fair, I don't think Zettlr's goal is a fully WYSIWYG, but something closer to a lightly pre-rendered
version of Markdown's [WYSIWYM](https://docs.zettlr.com/en/concepts/wysiwym/) model. But doesn't Zettlr's [preview window](https://docs.zettlr.com/en/core/print-preview/) provide some of what you're looking for? It doesn't update dynamically. I wonder, could it be changed to update whenever the file is saved?

**Reply by nathanlesage** [reply]
This. When I had to decide which type of Markdown editor Zettlr would become, I opted for WYSIWYM because a preview will never look like any export. Rather, what it serves, is to strip away the formatting symbols so that the code looks a little bit cleaner. However, we're almost there. Over the years, more and more Markdown formatting has received the option to be pre-rendered so that, by now, one could easily argue that it's almost WYSIWYG

With regard to a dynamically updating print preview, that should be possible, and the only Markdown characters that are requested to be pre-rendered are the italic and bold formatting, and then Zettlr is pretty much like Typora in that regard.

**Reply by RoyiAvital** [reply]
I only wish for RAW text mode. If we can have side by side preview window, great, if not, just being able to disable this auto rendering.

**Reply by nathanlesage** [reply]
@RoyiAvital You can disable every single piece of autorendering in the [Display preferences](https://docs.zettlr.com/en/reference/settings/#display). Unchecking all of these checkboxes will make Zettlr behave like a regular code editor with Syntax highlighting and without any pre-rendering.

**Reply by RoyiAvital** [reply]
@nathanlesage , Oh, I missed that.  
Now, we only miss the option to have a rendered view side by side :-).

**Reply by nathanlesage** [reply]
There was the issue that people want to quickly switch between all preview rendering and no preview rendering, which is I think also in your interest, even if it's not side-by-side, is it? Because I *really* don't want a second column in the main editor and rather do it in-place (kind of as Notable had this a few months ago when I last checked)

**Reply by RoyiAvital** [reply]
@nathanlesage , It could be a creative solution.
I'd say that being able to render inplace with a keyboard shortcut is almost equivalent.

I think that if you add an option to only render what's seen on the screen would be even better than a side by side since for large documents it will have better performance as the preview is only partial.

By the way, it is great to have [a portable version of this application](https://portableapps.com/apps/office/zettlr-portable) by @PortableApps.

**Reply by nathanlesage** [reply]
> I think that if you add an option to only render what's seen on the screen would be even better than a side by side since for large documents it will have better performance as the preview is only partial.

That is already happening: The app only renders what you see plus a few more lines in any case, which you will notice while very quickly scrolling through a long document.

**Reply by RoyiAvital** [reply]
@nathanlesage , That's great!
Where is the issue you mentioned?

I hope it will be implemented soon.

**Comment by thwaller** [comment]
I did not mean anything negative by my statements Everyone uses software differently and needs, wants and desires vary person to person. I have no issues with Zettlr at all, it is a good software and does a great job.

The preview windows is a nice option, but does not do what at least I am looking for. The preview is also not the only reason for my change, but one of them. No software will ever be 100% what a user wants, even if that user is the developer, so I never have unreasonable expectations with software, and always appreciate the work people do in regard to FOSS. I am and will be for sure interested in developments of Zettlr, so I hope all is taken properly.

**Comment by dj-sf** [comment]
just throwing in my 2 cents - I just discovered Zettlr and love the idea of it.  The only thing preventing me from using it instead of Obsidian is the lack of a "live preview" feature like what Obsidian has.  I know that not all users prefer this but it would be great to have it as an option.  

My reason is that I like to use Obsidian for brainstorming pretty often and having all the extra markdown formatting characters on display at all times is a bit jarring and distracts from the ideas that I am trying to get out of my head and into writing.

Basically, if Optional Live Preview was implemented, I would switch from Obsidian to Zettlr immediately and probably never look back.  If others feel similarly feel free to chime in.

**Reply by nathanlesage** [reply]
Try to activate all renderers in 3.0's display settings; I have finally budged in and implemented an *almost* live preview thing, very close actually to what Typora does. Maybe it's closer to what you're envisioning.

**Comment by ymic9963** [comment]
Commenting so that this feature gets added! It would be extremely helpful to have this.

**Reply by pastukhvladyslav** [reply]
Check **all** these settings (`Hide heading characters` and `Render emphasis` included)
![image](https://github.com/Zettlr/Zettlr/assets/25872559/400b8f22-d6e9-42a1-bd64-43f007a39ef2)

**Comment by IceflowRE** [comment]
Still a huge bummer this is not added.

**Reply by nathanlesage** [reply]
You mean proper live preview? This has been improved a lot, and I believe 4.0 does have that.

**Reply by IceflowRE** [reply]
This issue also mentions and asks for a split view, which iam not able to find. And there is no proper markdown split view editor out there. Even VSCode has huge problem with correctly syncing the source position with the rendering.

**Reply by nathanlesage** [reply]
Ah, I see. Well, this is indeed quite some amount of work and requires a lot of UX design, and so I can't promise whether or if this will ever come.
